adyenv6notificationv2/src/com/adyen/v6/security/AdyenNotificationAuthenticationProvider.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *                        ######
+ *                        ######
+ *  ############    ####( ######  #####. ######  ############   ############
+ *  #############  #####( ######  #####. ######  #############  #############
+ *         ######  #####( ######  #####. ######  #####  ######  #####  ######
+ *  ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ *  ###### ######  #####( ######  #####. ######  #####          #####  ######
+ *  #############  #############  #############  #############  #####  ######
+ *   ############   ############  #############   ############  #####  ######
+ *                                       ######
+ *                                #############
+ *                                ############
+ *
+ *  Adyen Hybris Extension
+ *
+ *  Copyright (c) 2017 Adyen B.V.
+ *  This file is open source and available under the MIT license.
+ *  See the LICENSE file for more info.
+ */
+package com.adyen.v6.security;
+
+import com.adyen.model.notification.NotificationRequest;
+import com.adyen.model.notification.NotificationRequestItem;
+import com.adyen.util.HMACValidator;
+import de.hybris.platform.basecommerce.model.site.BaseSiteModel;
+import de.hybris.platform.site.BaseSiteService;
+import de.hybris.platform.store.BaseStoreModel;
+import de.hybris.platform.store.services.BaseStoreService;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import java.nio.charset.Charset;
+import java.security.SignatureException;
+import java.util.Base64;
+
+/**
+ * Authenticates a request that is using Basic Authentication
+ */
+@Component
+public class AdyenNotificationAuthenticationProvider {
```

Comment:
This class is duplication of the same class in adyenv6noficiation. Please move it to core and use one implementation in both extensions.